source/lac/scalapack.cc

```diff
@@ -1038,6 +1038,45 @@ void ScaLAPACKMatrix<NumberType>::least_squares(ScaLAPACKMatrix<NumberType> &B,
 
 
 
+template <typename NumberType>
+unsigned int ScaLAPACKMatrix<NumberType>::pseudoinverse(const NumberType ratio)
+{
+  Assert(state == LAPACKSupport::matrix,
+         ExcMessage("Matrix has to be in Matrix state before calling this function."));
+  Assert(row_block_size==column_block_size,
+         ExcMessage("Use identical block sizes for rows and columns of matrix A"));
+
+  ScaLAPACKMatrix<NumberType> U(n_rows,n_rows,grid,row_block_size,row_block_size,LAPACKSupport::Property::general);
+  ScaLAPACKMatrix<NumberType> VT(n_columns,n_columns,grid,row_block_size,row_block_size,LAPACKSupport::Property::general);
+  std::vector<NumberType> sv = compute_SVD(&U,&VT);
+  AssertThrow(sv[0]>1e-16,ExcMessage("Matrix has rank 0"));
+
+  // Get number of singular values fulfilling the following: sv[i] > sv[0] * ratio
+  unsigned int n_sv=1;
+
+  for (unsigned int i=1; i<sv.size(); ++i)
+    if (sv[i] > sv[0] * ratio)
+      ++n_sv;
+    else break;
+
+  std::vector<NumberType> inv_sigma(n_sv);
+  for (unsigned int i=0; i<n_sv; ++i)
+    inv_sigma[i] = 1/sv[i];
+
+  ScaLAPACKMatrix<NumberType> U_R(n_rows,n_sv,grid,row_block_size,row_block_size,LAPACKSupport::Property::general);
```

Comment:
can't you use `U` and `VT` directly? they are anyway temporary internal variables

Reply:
See comment!